core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.ts

```diff
@@ -276,6 +279,18 @@ export class LessonInformationCardModalComponent extends ConfirmOrCancelModal {
     });
   }
 
+  returnToCheckpoint(index: number): void {
+    const cardIndex = this.checkpointCardsIndex[index];
+    if (cardIndex !== undefined) {
```

Comment:
A very good rule of thumb - prefer handling corner cases or errors before handling the general case.

Here
```
if (error) {
  log()
  return
}
<general_case>
```

The less nesting (if-s/for-s/etc) you have the better, if you have a code that is placed under if-for-if-for, one needs to keep all those items in their head, and that's not too simple

Reply:
Thank you for the suggestion! Corrections made.